src/github.com/getlantern/detour/detour.go

```diff
@@ -213,12 +213,12 @@ func (c *conn) Write(b []byte) (int, error) {
 
 func (c *conn) Close() error {
 	close(c.closed)
-	c.detourAllowed.Stop()
 	_ = c.direct.Close()
 	_ = c.detour.Close()
 
 	log.Tracef("%s: Should detour? %v - Detourable? %v", c.addr, c.direct.ShouldDetour(), c.detour.Detourable())
 	allowed, valid := c.detourAllowed.Get(0)
+	c.detourAllowed.Cancel()
```

Comment:
@fffw Due to the slightly different semantics of Cancel, I moved this to after I check detourAllowed.